api4/post.go

```diff
@@ -576,6 +577,30 @@ func patchPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(patchedPost.ToJson()))
 }
 
+func setPostUnread(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+	if c.App.Session.UserId != c.Params.UserId {
```

Comment:
As I mentioned in the API review, we're going to want to support this behaviour. Instead of returning a permission error here, I think we just need to check that the session has that permission instead.

Reply:
ok, I think I missanderstood you then. I'll change it :)